src/back/spv/writer.rs

```diff
@@ -811,13 +827,49 @@ impl Writer {
                         .unwrap();
                     constituent_ids.push(component_id);
                 }
+                let constituent_ids_slice = constituent_ids.as_slice();
+
+                let id = match ir_module.types[*ty].inner {
+                    crate::TypeInner::Vector { .. } => {
+                        self.write_composite_construct(base_type_id, constituent_ids_slice, block)
+                    }
+                    crate::TypeInner::Matrix {
+                        rows,
+                        columns,
+                        kind,
+                        width,
+                    } => {
+                        let vector_type_id = self.get_type_id(
+                            &ir_module.types,
+                            LookupType::Local(LocalType::Vector {
+                                width,
+                                kind,
+                                size: columns,
+                            }),
+                        );
+
+                        let capacity = match rows {
+                            crate::VectorSize::Bi => 2,
+                            crate::VectorSize::Tri => 3,
+                            crate::VectorSize::Quad => 4,
+                        };
+
+                        let mut vector_ids = Vec::with_capacity(capacity);
+
+                        for _ in 0..capacity {
+                            let vector_id = self.write_composite_construct(
+                                vector_type_id,
+                                constituent_ids_slice,
+                                block,
+                            );
+                            vector_ids.push(vector_id);
+                        }
+
+                        self.write_composite_construct(base_type_id, vector_ids.as_slice(), block)
+                    }
+                    _ => unreachable!(),
```

Comment:
Cases like this should actually `panic` instead of being `unreachable`. The condition to fail here is basically working on an IR that wasn't validated.
Unreachable should be used where some of the other code *has* to ensure that this is good, so it's truly no ever expected to come there.